src/components/dashboard/Send.js

```diff
@@ -47,15 +47,16 @@ const validate = to => {
 
 const ContinueButton = ({ screenProps, to, disabled, checkError }) => (
   <CustomButton
-    onPress={() => {
+    onPress={async () => {
       if (checkError()) return
 
       if (to && (isMobilePhone(to) || isEmail(to))) {
         return screenProps.push('Amount', { to, nextRoutes: ['Reason', 'SendLinkSummary'] })
       }
 
       if (to && goodWallet.wallet.utils.isAddress(to)) {
-        return screenProps.push('Amount', { to, nextRoutes: ['Reason', 'SendQRSummary'] })
+        const profile = await UserStorage.getUserProfile(to)
```

Comment:
Why are we getting the profile here and being passing it around instead of getting it on the summary?

Reply:
It's because I have a problem when I try to use await to get profile in the same render action...it throws an error about: `Invariant Violation: Objects are not valid as a React child`...for that reason I finally added in send file.

Reply:
@kbardi you would need to use state in order to do that. in did its not possible to use await inside render. discuss with dario and fer on best approach